tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+
+# cert-manager install process adapted from:
+# https://docs.cert-manager.io/en/latest/getting-started/install/kubernetes.html#installing-with-regular-manifests
+- name: Create namespace for cert-manager
+  k8s:
+    context: "{{ k8s_context }}"
+    kubeconfig: "{{ k8s_kubeconfig }}"
+    wait: yes
+    validate:
+      fail_on_error: yes
+      strict: yes
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: cert-manager
+  register: cert_manager
+
+- name: Temporarily disable validation for cert-manager namespace
+  k8s:
+    context: "{{ k8s_context }}"
+    kubeconfig: "{{ k8s_kubeconfig }}"
+    wait: yes
+    validate:
+      fail_on_error: yes
+      strict: yes
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: cert-manager
+        labels:
+          certmanager.k8s.io/disable-validation: "true"
+  when: cert_manager is changed
+
+- name: Deploy cert-manager and Let's Encrypt
+  k8s:
+    context: "{{ k8s_context }}"
+    kubeconfig: "{{ k8s_kubeconfig }}"
+    wait: yes
+    # FIXME: cert-manager.yaml doesn't validate, so this is disabled for now.
+    # Maybe the next version will validate?
+    # validate:
+    #   fail_on_error: yes
+    #   strict: yes
```

Comment:
This is worrying...   maybe add the comment from the instructions to clarify?

"Note: If you are running Kubernetes v1.15 or below, you will need to add the --validate=false flag to your kubectl apply command above else you will receive a validation error relating to the x-kubernetes-preserve-unknown-fields field in cert-manager’s CustomResourceDefinition resources. This is a benign error and occurs due to the way kubectl performs resource validation."

Reply:
I'm responsible for this originally and don't recall the exact details, but I think the `kubectl` validation may be different than the Python `kubernetes-validate` validation. Let's see how this goes with the next upgrade and comment it well at that time.